storage/entities/peer.js

```diff
@@ -0,0 +1,304 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const { defaults, omit, pick } = require('lodash');
+const filterType = require('../utils/filter_types');
+const { stringToByte } = require('../utils/inputSerializers');
+const BaseEntity = require('./base_entity');
+
+const defaultCreateValues = {};
+
+const readOnlyFields = [];
+
+/**
+ * Peer
+ * @typedef {Object} Peer
+ * @property {number} id
+ * @property {string} ip
+ * @property {number} wsPort
+ * @property {number} state
+ * @property {string} os
+ * @property {string} version
+ * @property {number} clock
+ * @property {string} broadhash
+ * @property {number} height
+ */
+
+/**
+ * Peer Filters
+ * @typedef {Object} filters.Peer
+ * @property {number} [id]
+ * @property {number} [id_eql]
+ * @property {number} [id_ne]
+ * @property {number} [id_gt]
+ * @property {number} [id_gte]
+ * @property {number} [id_lt]
+ * @property {number} [id_lte]
+ * @property {number} [id_in]
+ * @property {string} [ip]
+ * @property {string} [ip_eql]
+ * @property {string} [ip_ne]
+ * @property {string} [ip_in]
+ * @property {string} [ip_like]
+ * @property {number} [wsPort]
+ * @property {number} [wsPort_eql]
+ * @property {number} [wsPort_ne]
+ * @property {number} [wsPort_gt]
+ * @property {number} [wsPort_gte]
+ * @property {number} [wsPort_lt]
+ * @property {number} [wsPort_lte]
+ * @property {number} [wsPort_in]
+ * @property {number} [state]
+ * @property {number} [state_eql]
+ * @property {number} [state_ne]
+ * @property {number} [state_gt]
+ * @property {number} [state_gte]
+ * @property {number} [state_lt]
+ * @property {number} [state_lte]
+ * @property {number} [state_in]
+ * @property {string} [os]
+ * @property {string} [os_eql]
+ * @property {string} [os_ne]
+ * @property {string} [os_in]
+ * @property {string} [os_like]
+ * @property {string} [version]
+ * @property {string} [version_eql]
+ * @property {string} [version_ne]
+ * @property {string} [version_in]
+ * @property {string} [version_like]
+ * @property {number} [clock]
+ * @property {number} [clock_eql]
+ * @property {number} [clock_ne]
+ * @property {number} [clock_gt]
+ * @property {number} [clock_gte]
+ * @property {number} [clock_lt]
+ * @property {number} [clock_lte]
+ * @property {number} [clock_in]
```

Comment:
I don't think we are using `clock` attribute to filter peers any where. Please double check the code and if not found remove this filter as well.

Reply:
IMO we should provide filters for all attributes as it can be used by 3rd party applications.

Reply:
I do agree that all attributes should have filters. Resolving for now.